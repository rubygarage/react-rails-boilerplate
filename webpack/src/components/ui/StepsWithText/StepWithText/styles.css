@media (min-width: 768px) {
  .step {
    min-width: 10rem;
    display: inline-block;
    font-size: 1.2rem;
    padding: 0.5rem 0.3rem;
    color: #6d6d6d;
    position: relative;

    &.active {
      color: white;
    }

    &:before {
      content: '';
      height: 0.7rem;
      width: 0.7rem;
      border-radius: 50%;
      background-color: currentColor;
      position: absolute;
      bottom: 0;
      left: 50%;
      transform: translate(-50%, 100%);
    }

    &:after {
      content: '';
      height: 0.1rem;
      width: 80%;
      background-color: rgba(255, 255, 255, 0.2);
      position: absolute;
      bottom: -0.3rem;
      left: 60%;
      transform: translate(0%, 100%);
    }

    &:last-of-type:after {
      content: none;
    }
  }

  .active {
    &.step {
      color: white;
    }

    .step-text {
      color: white;
    }
  }

  .passed {
    &.step {
      color: #637fa0;
    }

    .step-text {
      color: #6d6d6d;
    }
  }
}

@media (min-width: 1200px) {
  .step {
    min-width: 11rem;
  }
}
