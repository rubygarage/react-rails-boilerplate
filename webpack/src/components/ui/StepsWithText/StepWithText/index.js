import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './styles.css'

function StepWithText({ text, active, passed }) {
  return (
    <div className={classNames(styles.step, { [styles.active]: active, [styles.passed]: passed })}>
      <span className={styles['step-text']}>
        {text}
      </span>
    </div>
  )
}

StepWithText.propTypes = {
  active: PropTypes.bool,
  passed: PropTypes.bool,
  text: PropTypes.string
}

export default StepWithText
