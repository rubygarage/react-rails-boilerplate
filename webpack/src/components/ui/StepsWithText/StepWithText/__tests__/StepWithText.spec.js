import React from 'react'
import { shallow } from 'enzyme'
import StepWithText from 'components/ui/StepsWithText/StepWithText'

describe('<StepWithText />', () => {
  const defaultProps = {
    active: true,
    passed: false,
    text: 'text'
  }

  it('renders active step', () => {
    const component = shallow(
      <StepWithText {...defaultProps} />
    )

    expect(component).toMatchSnapshot()
  })

  it('renders passed step', () => {
    const props = {
      ...defaultProps,
      active: false,
      passed: true
    }

    const component = shallow(
      <StepWithText {...props} />
    )

    expect(component).toMatchSnapshot()
  })
})
