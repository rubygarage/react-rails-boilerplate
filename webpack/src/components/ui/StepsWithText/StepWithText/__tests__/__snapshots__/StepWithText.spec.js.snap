// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepWithText /> renders active step 1`] = `
<div
  className="step active"
>
  <span
    className="step-text"
  >
    text
  </span>
</div>
`;

exports[`<StepWithText /> renders passed step 1`] = `
<div
  className="step passed"
>
  <span
    className="step-text"
  >
    text
  </span>
</div>
`;
